File: falldownview/src/main/java/com/mitsuki/falldownview/path/ComponentSample.java
package com.mitsuki.falldownview.path;

import android.graphics.Matrix;
import android.graphics.Path;

public class ComponentSample implements ComponentPath {
    protected float baseline;

    protected Path mPath;

    @Override
    public int getBaseLine() {
        return (int) baseline;
    }

    @Override
    public Path getObjPath(float size) {
        Path path = new Path(mPath);
        Matrix matrix = new Matrix();
        matrix.setScale(size / baseline, size / baseline);
        path.transform(matrix);
        return path;
    }
}

File: falldownview/src/main/java/com/mitsuki/falldownview/path/ComponentSample.kt
package com.mitsuki.falldownview.path

import android.graphics.Matrix
import android.graphics.Path

abstract class ComponentSample : ComponentPath {

    protected lateinit var mPath: Path

    open val defaultScale: Float = 1f

    override fun particlePath(size: Float): Path {
        val path = Path(mPath)
        val matrix = Matrix()
        matrix.setScale(size / sizeBenchmark * defaultScale, size / sizeBenchmark * defaultScale)
        path.transform(matrix)
        return path
    }

}
